### ISSUE TITLE
Warning: rename(,.html): Invalid argument in ...

### ISSUE THREAD

**Issue by luis-alves** [mikehaertl/phpwkhtmltopdf#198]
Hi. For the past days the phpwkhtmltopdf worked well, but all the sudden it stopped working! giving mainly errors in this two lines 47 and 59, with the rename and unlink function.  This is my setup:  
  
     <?php
    require '../vendor/autoload.php';
    
    //echo \mikehaertl\tmp\File::getTempDir();
    
    $template = $_GET['printit'];
    //d($template);
    use mikehaertl\wkhtmlto\Pdf;
    
    ob_start();
    require $template;
    $content = ob_get_clean();
    
    // You can pass a filename, a HTML string, an URL or an options array to the constructor
    $pdf = new Pdf($content);

    $pdf->tmpDir = '../tmp/';
    $pdf->binary = 'C:\Program Files\wkhtmltopdf\bin\wkhtmltopdf.exe';
    $pdf -> setOptions(['orientation' => 'Landscape']);
    // On some systems you may have to set the path to the wkhtmltopdf executable
    
    $pdf->send('teste.pdf');
    
    ?>  
  
I can see the temp file being created to soon after it disappears and the following error is displayed:  
  
    Warning: rename(,.html): Invalid argument in C:\Users\...\vendor\mikehaertl\php-tmpfile\src\File.php on line 47

    Warning: unlink(.html): No such file or directory in C:\Users\...\vendor\mikehaertl\php-tmpfile\src\File.php on line 59  
  
What even weird it that occasionally it creates the pdf, just by trying several times, and then it stops working!   
What could be the problem?

**Comment by mikehaertl** [comment]
My guess is: It has something to do with the Windows environment. Unfortunately I never use Windows for development so I can't help much with debugging.

You've already looked at `vendor/mikehaertl/php-tmpfile/src/File.php`, so I would inspect this further. The file is renamed there to add a `.pdf` suffix. You should check, which filenames are used there, and why there's no filename in the error message. The logic in the full file should not be too hard to follow so maybe you can do some simple tests without phpwkhtmltopdf, using only tmpfile.

If you find an issue (and maybe a fix), please open an issue over at the tmpfile project: https://github.com/mikehaertl/php-tmpfile

**Comment by luis-alves** [comment]
The temporary file that it creates is something like: `tempJE2O.tmp.pdf`, so I think it creates it well.

**Comment by mikehaertl** [comment]
Ok, but the error message above shows an empty filename. You somehow need to find out, how the filename can suddenly be empty.

**Comment by luis-alves** [comment]
Ok got it fixed, the IT guy ways! Restarted Windows and its executing well again. Sorry for the inconvenience. And good job with this library. Thanks!
